src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -1228,15 +1262,21 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
         default: break;
       }
 
-      if (thread->second.retry_oom_injected > 0) {
-        thread->second.retry_oom_injected--;
-        thread->second.metrics.num_times_retry_throw++;
-        log_status("INJECTED_RETRY_OOM", thread_id, thread->second.task_id, thread->second.state);
-        thread->second.record_failed_retry_time();
-        if (is_for_cpu) {
-          throw_java_exception(CPU_RETRY_OOM_CLASS, "injected RetryOOM");
-        } else {
-          throw_java_exception(GPU_RETRY_OOM_CLASS, "injected RetryOOM");
+      if (thread->second.retry_oom.matches(is_for_cpu)) {
+        if (thread->second.retry_oom.skip_count > 0) {
+          logger->info("SKIPPING INJECTED_RETRY_OOM({}) thread_id={},task_id={},skip_count={}",
```

Comment:
nit: The logger was setup with a very specific pattern to make it so we could parse it. This is not that hard to filter out, but why not just have it do a log_status? or change it to debug like the other one.